File: Course/712 Testing/HW2/Product.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;

namespace Sample
{
    class Product
    {
        private String title;
        private int price;

        public Product(String title, int price) 
        {
            this.title = title;
            this.price = price;
        }

        public String getTitle() 
        {
            return title;
        }

        public int getPrice() 
        {
            return price;
        }

        override public bool Equals(Object o) 
        {
            if (o.GetType()== this.GetType()) 
            {
                Product p = (Product)o;
                return p.getTitle().Equals(title);
            }
            return false;
        }

        override public int GetHashCode()
        {
            return title.GetHashCode() + price;
        }
    
   }
}

File: trunk/Course/712 Testing/HW2/junitprimer_src/Product.java
/**
 * An example product for use in the shopping cart
 * as described in the JUnit Primer.
 *
 * @author Mike Clark, http://clarkware.com
 */
 
public class Product {

    private String title;
    private double price;
    
    public Product(String title, double price) {
	this.title = title;
	this.price = price;
    }
    
    public String getTitle() {
	return title;
    }
    
    public double getPrice() {
	return price;
    }
    
    public boolean equals(Object o) {
	if (o instanceof Product) {
	    Product p = (Product)o;
	    return p.getTitle().equals(title);
	}
	
	return false;
    }
}
